libpromises/evalfunction.c

```diff
@@ -5461,6 +5461,8 @@ static FnCallResult FnCallReadFile(ARG_UNUSED EvalContext *ctx, ARG_UNUSED const
     }
     else
     {
+        Log(LOG_LEVEL_ERR, "Function '%s' failed to read file: %s",
+            fp->name, filename);
         return FnFailure();
```

Comment:
I have no strong opinion, but only

>  Redmine #7460: Unable to edit the same value in multiple regions. 

looks important here. Other stuff is changing the behavior as well so I am not sure if we should add it in 3.6. @jimis what do you think about filesafe generation and interactive tty?